File: time/py/duplicate-times.py
#!/usr/bin/python3

import sys

def main(argv):

    n = int (argv[0], 16)
    times = int (argv[1])

    for i in range(times):
        n = n + n
        
    print (hex (n))


if __name__ == '__main__':
    if len(sys.argv) != 3:
        print("usage: add NUM NUM")
        sys.exit(0)

    else:
        main(sys.argv[1:])

File: time/duplicate-times.cpp
#include <iostream>
#include <natural.hpp>
#include <string>

int main(int argc, char ** argv)
{
    if (argc != 3) {
        std::cout << "Usage: add-times NUMBER TIMES" << std::endl;
        return 0;
    }
       
    lint::natural natural{std::string{argv[1]}};
    int times{std::stoi(argv[2])};
       
    std::cout << natural << "  num: " << argv[1]
              << "\t times: 0x" << times
              << std::endl;
    for (int i = 0 ; i < times; i++) {
        natural += natural;
    }
    std::cout << natural << std::endl;
}
